### ISSUE TITLE
Consider removal of @store annotation in its current form

### ISSUE THREAD

**Issue by shyndman** [mobxjs/mobx.dart#347]
(Note that this discussion is orthogonal to redefining `@store` for other purposes, as discussed in #219)

This refers to the style of declaring a store like this:
```dart
@store
class _User {
  @observable
  String firstName = 'Jane';
}
```
and having mobx generate a public subclass, `User`, which sprinkles in MobX observability.

This approach is flawed for two reasons:
1. Only the private type is available for annotation, which leads to incompatibilities in some other part generators (see #147) which require knowledge of the public type.
2. The public type is generated, and as such, is inaccessible to the build system's analyzer. This means that when that store's public type is referenced elsewhere, the builders will not be able to resolve information about it.

We've been able to get around 2 in our use cases by pulling out names from source exactly as the user wrote them, but this is increasingly becoming cumbersome (see #324), preventing us from leaning on the analyzer for its understanding of the code. 

I recommend we remove `@store` in its current form. It's a source of problems, and I expect them to grow as time goes on.

**Comment by shyndman** [comment]
I wrote a PR to remove support for `@store` in the code generator, but I'm going to leave it in draft for a few days to see whether people have comments.

**Comment by shyndman** [comment]
Closed via #349
